src/bp-messages/classes/class-bp-messages-thread.php

```diff
@@ -1317,10 +1316,22 @@ public static function get_inbox_count( $user_id = 0 ) {
 		$unread_count = wp_cache_get( $user_id, 'bp_messages_unread_count' );
 
 		if ( false === $unread_count ) {
-			$bp = buddypress();
 
 			// $unread_count = (int) $wpdb->get_var( $wpdb->prepare( "SELECT SUM(unread_count) FROM {$bp->messages->table_name_recipients} WHERE user_id = %d AND is_deleted = 0 AND sender_only = 0", $user_id ) );
-			$unread_count = (int) $wpdb->get_var( $wpdb->prepare( "SELECT SUM(unread_count) FROM {$bp->messages->table_name_recipients} WHERE user_id = %d AND is_deleted = 0", $user_id ) ); // WPCS: db call ok.
+			//$unread_count  = (int) $wpdb->get_var( $wpdb->prepare( "SELECT SUM(unread_count) FROM {$bp->messages->table_name_recipients} WHERE user_id = %d AND is_deleted = 0", $user_id ) ); // WPCS: db call ok.
+			$unread_counts = BP_Messages_Thread::get(
+				array(
+					'user_id'    => $user_id,
+					'per_page'   => - 1,
```

Comment:
@janakkaneriya007 Why we need to pass `per_page => -1`

Reply:
@dipskakadiya We would need per_page => -1 to sum all the unread count for the particular user.